src/commonMain/kotlin/fr/acinq/eclair/channel/Channel.kt

```diff
@@ -164,7 +167,41 @@ sealed class ChannelState {
                 this is WaitForInit && newState is Closing -> actions
                 this is Closing && newState is Closing -> actions
                 this is ChannelStateWithCommitments && newState is Closing -> {
-                    actions + ChannelAction.Storage.StoreOutgoingAmount(this.commitments.localCommit.spec.toLocal)
+                    val channelBalance = commitments.localCommit.spec.toLocal
```

Comment:
Note that it can quickly become tricky though if we want to do it *perfectly*, but we can use a simple approximation:

- we can consider that all incoming HTLCs will be claimed by us (this ignores only the case where people send us payments for invoices that aren't ours, which shouldn't happen often)
- we can consider that all outgoing HTLCs will work and correctly reach the recipient who will fulfill them, so we ignore them in our balance calculation

Reply:
> it can quickly become tricky though if we want to do it _perfectly_

The goal is to "balance the ledger" from the user's point-of-view. The user had a balance of X in the channel. And then the channel was closed. So the transaction should have an amount of -X

I used `commitments.localCommit.spec.toLocal` for the simple reason that, it's what we use in phoenix-kmm to report the user's balance (per channel). [link](https://github.com/ACINQ/phoenix-kmm/blob/d555efe981cd37149578bf2091d5fedf21d7bb98/phoenix-shared/src/commonMain/kotlin/fr.acinq.phoenix/app/ctrl/AppHomeController.kt#L48)

Reply:
You're right, it's probably good enough to only look at the main output, either way could be confusing in some scenarios so let's keep the simplest one.